pkg/runtime/scheduler/scheduler.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	schedulerv1pb "github.com/dapr/dapr/pkg/proto/scheduler/v1"
+	"github.com/dapr/dapr/pkg/scheduler"
+	schedulerclient "github.com/dapr/dapr/pkg/scheduler/client"
+	"github.com/dapr/dapr/pkg/security"
+	"github.com/dapr/kit/logger"
+	"google.golang.org/grpc"
+)
+
+var log = logger.NewLogger("dapr.runtime.scheduler")
+
+type Scheduler struct {
+	Addresses    []string
+	SidecarAddr  string
+	SidecarPort  int
+	SidecarNS    string
+	SidecarAppID string
+	Sec          security.Handler
+}
+
+// Client represents a client for interacting with the scheduler.
+type Client struct {
+	conn       *grpc.ClientConn
+	scheduler  schedulerv1pb.SchedulerClient
+	address    string
+	secHandler security.Handler
+}
+
+// Manager manages connections to multiple schedulers.
+type Manager struct {
+	Clients     []*Client
+	ConnDetails scheduler.SidecarConnDetails
+	Lock        sync.Mutex
+	LastUsedIdx int
+	wg          sync.WaitGroup
+}
+
+func NewManager(ctx context.Context, sched Scheduler) (*Manager, error) {
+	manager := &Manager{
+		ConnDetails: scheduler.SidecarConnDetails{
+			Namespace: sched.SidecarNS,
+			Host:      sched.SidecarAddr,
+			Port:      sched.SidecarPort,
+			AppID:     sched.SidecarAppID,
+		},
+	}
+
+	for _, address := range sched.Addresses {
+		conn, client, err := schedulerclient.New(ctx, address, sched.Sec)
+
+		if err != nil {
+			return nil, fmt.Errorf("error creating scheduler client for address %s: %w", address, err)
+		}
+		log.Infof("Scheduler client initialized for address: %s\n", address)
+
+		manager.Clients = append(manager.Clients, &Client{
+			conn:       conn,
+			scheduler:  client,
+			address:    address,
+			secHandler: sched.Sec,
+		})
+	}
+
+	return manager, nil
+}
+
+// Run starts watching for job triggers from all scheduler clients.
+func (m *Manager) Run(ctx context.Context) {
```

Comment:
This function should be blocking, and returns an error returned from all `watchJob` calls.